api/users/serializers.py

```diff
@@ -117,6 +123,25 @@ def get_absolute_url(self, obj):
             'version': self.context['request'].parser_context['kwargs']['version']
         })
 
+    def get_node_count(self, obj):
+        return default_node_list_queryset(model_cls=Node).filter(contributor__user__id=obj.id).count()
+
+    def get_quickfiles_count(self, obj):
+        return QuickFilesNode.objects.get(contributor__user__id=obj.id).files.filter(type='osf.osfstoragefile').count()
+
+    def get_registration_count(self, obj):
+        return default_node_list_queryset(model_cls=Registration).filter(contributor__user__id=obj.id).count()
+
+    def get_preprint_count(self, obj):
+        return PreprintService.objects.filter(Q(node___contributors__guids___id=obj._id)
```

Comment:
It looks to me like deleted preprints aren't accounted for here, there's a confusion between the logged in user and the target user, and reviewers will see a different count.

The UserPreprints queryset starts with a base queryset of preprints who have the target user as a contributor. That base queryset is further filtered to make sure preprints are not deleted and to see if the logged in user is either an admin/reviewer or the preprint is viewable by anyone (published/public).

That being said, I would make it something like this:
```python
from guardian.shortcuts import get_objects_for_user
from osf.models import PreprintProvider
    class UserSerializer(JSONAPISerializer):
    ...
      def get_preprint_count(self, obj):
          auth_user =  get_user_auth(self.context['request']).user
          query = Q(is_published=True, node__is_public=True)
          if auth_user:
              admin_user_query = Q(node__contributor__user_id=auth_user.id, node__contributor__admin=True)
              reviews_user_query = Q(node__is_public=True, provider__in=get_objects_for_user(auth_user, 'view_submissions', PreprintProvider))
              query = query | admin_user_query | reviews_user_query

          return PreprintService.objects.filter(node___contributors__guids__id=obj.id, node__is_deleted=False).filter(query).count()
```

This is very similar to logic in PreprintFilterMixin.preprints_queryset but I wouldn't waste your time trying to consolidate because the divorce rewrites and moves that code.  More of a note to self: if your PR gets in before mine, divorce code will need to tweak your counts.